nistats/second_level_model.py

```diff
@@ -436,12 +433,17 @@ def compute_contrast(
         contrast = mem_contrast(self.labels_, self.results_, con_val,
                                 second_level_stat_type)
 
-        # We get desired output from contrast object
-        estimate_ = getattr(contrast, output_type)()
+        output_types = valid_types[:-1] if output_type == 'all' else [output_type]
+
+        outputs = {}
+        for output_type_ in output_types:
+            # We get desired output from contrast object
```

Comment:
@bthirion By inserting `assert False` here, I've managed to show the following effect in PDB:

```Python
ipdb> contrast.p_value().shape
(256,)
ipdb> contrast.stat().shape                                                                         
(256,)
ipdb> contrast.p_value().shape                                                                      
(1, 256)
```

This is reliable. If y'all have time to look at what the `contrast` object is doing where the `stat` method has a side-effect of changing the shapes of the return values of the `p_value` and `z_score` methods, I think that's the heart of the problem. I can try digging more later.